### DISCUSSION TITLE
Can I use Transmission logo as my browser extension icon?

### DISCUSSION THREAD

**Discussion by hanguokai** [transmission/transmission#4105] [General]
Hi, I would like to create a browser extension: when users right click a magnet link, then add it to transmission remote via rpc api.

My question is can I use Transmission logo as this extension icon? This [file](https://github.com/transmission/transmission/blob/main/web/assets/img/README.md) says the logo is under the MIT license.

**Comment by mikedld** [comment]
Do you really need an extension for that? I think #2874 takes care of your use-case, unless there's more to it.

**Comment by GaryElshaw** [comment]
This is out there: https://chrome.google.com/webstore/detail/transmitter-for-transmiss/cdmpmfcgepijfiaaojbahpmpjfkgdgja

**Comment by hanguokai** [comment]
@mikedld Using `registerProtocolHandler()` is a useful improvement for built-in Web UI. It seems a new feature in 4.0, I didn't found it before (at present, I use 3.0). [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/registerProtocolHandler) doc says `registerProtocolHandler` is available only in secure contexts (HTTPS). Therefore, it can't work like "http://ip:9091/transmission/web/". So a browser extension is still useful for many people, and it can supply more features with extension api + rpc api.

If the extension icon is the same as transmission icon, users know its function at a glance, and I don't need to design a new icon.

@GaryElshaw I know there are some existing extensions that supply this function. But most of them require "Read and change all your data on all websites." permission. I am an extension developer, I can create a simple extension without that permission.

**Reply by GaryElshaw** [reply]
I have that set for 'On click', there's no need for it to access everything.

**Comment by mikedld** [comment]
As to the original question, the PR that added that README file (#1476) specified MIT as the license for the icon that was there back then [1]. Another PR that updated the icon (#1650 + #1653) to its current form [2] mentions Rodger Werner as its author (no contact info provided), so I'm not sure if the same license still applies. CC @livings124

[1] ![Old icon](https://github.com/transmission/transmission/raw/77b44d5d88d5e008b2d774e988a6cba2eedc7776/web/style/images/logo.png)

[2] ![Current icon](https://github.com/transmission/transmission/raw/7edd656536af0e851905a1171ee1916af9c60904/macosx/Images/Images.xcassets/AppIcon.appiconset/icon_32x32%402x.png)

**Reply by hanguokai** [reply]
Thanks for the explanation. I will use it after you confirm it, or use another icon instead.

**Comment by hanguokai** [comment]
Any updates about new icon license? My first version of the extension is almost done. Not only a right-click add function, it contains a basic web ui for transmission.

**Comment by hanguokai** [comment]
I just published my extension. Currently I use the old Transmission icon as the extension icon. At the bottom of the extension description, I declare that the extension icon is from official Transmission under MIT licence.

Install Links: [Chrome](https://chrome.google.com/webstore/detail/hfhhlbapamidaibjcgdmeidlcoaoadme), [Edge](https://microsoftedge.microsoft.com/addons/detail/pefbobclhiokohhohjobgiaelbdcnaeb)

**Comment by ckerr** [comment]
That extension looks great! :heart: 

My main concern is that with the Transmission logo + the name being "Transmission Control" that it looks like it's a release from the Transmission project. If Transmission's name was in a dependent clause I'd be happier, e.g. "Control for Transmission".

OTOH there have been other projects over the years calling themselves themselves `Transmission ${foo}` that never bothered to ask for permission and I don't want to penalize you for asking permission :smile: If you _really_ prefer "Transmission Control" then go ahead as long as you have a prominent disclaimer that it's not a transmissionbt project.

**Reply by hanguokai** [reply]
1. Changing name to "Control for Transmission": Ok
2. Adding a disclaimer: Ok

I will update it in a week. (This needs to publish a version and wait for extension store's review)

**Comment by hanguokai** [comment]
The extension has been updated.

- Icon: I designed a new icon by myself, a simple graph.
- Name: "Control for Transmission"
- Disclaimer: I added this statement at the bottom of the description (click "read more"): 

> "This extension is developed and implemented by the extension author based on the RPC API provided by Transmission. It's not a transmissionbt project."

![icon](https://user-images.githubusercontent.com/776467/208840471-a4a62fcd-9964-48a1-97bb-7b918a2c91dc.png)
